cdci_data_analysis/analysis/renku_helper.py

```diff
@@ -170,18 +172,42 @@ def create_new_notebook_with_code(repo, api_code, job_id, file_name=None):
     return file_path
 
 
-def commit_and_push_file(repo, file_path):
-    try:
-        add_info = repo.index.add(file_path)
-        commit_info = repo.index.commit("commit code from MMODA")
-        origin = repo.remote(name="origin")
-        # TODO make it work with methods from GitPython
-        # e.g. push_info = origin.push(refspec='origin:' + str(repo.head.ref))
-        push_info = repo.git.push("--set-upstream", repo.remote().name, str(repo.head.ref))
-        logger.info("push operation complete")
-    except Exception as e:
-        logger.warning(f"something happened while pushing the the file {file_path}, {e}")
-        raise e
+def generate_commit_request_url(params_dic, products_url):
+    # generate the url for the commit message
+    # this is a "default" value for use_scws
+    params_dic['use_scws'] = 'no'
```

Comment:
is this logic not also applicable to the url in email?

Reply:
yes, but for the email the instance member `self.use_scws` is used:

[https://github.com/oda-hub/dispatcher-app/blob/9ad3167723a116e4b485606feaf7c90dbc209075/cdci_data_analysis/flask_app/dispatcher_query.py#L1019-L1026](https://github.com/oda-hub/dispatcher-app/blob/9ad3167723a116e4b485606feaf7c90dbc209075/cdci_data_analysis/flask_app/dispatcher_query.py#L1019-L1026)

Reply:
so maybe we can first insert the value from the member in `par_dict`, and then do the logic inside the generate_url function?

Reply:
that's just annoying, since it's a delicate logic and it's unfortunate to have it twice.
ok, well.